processor/src/advice/mem_provider.rs

```diff
@@ -31,6 +30,16 @@ impl From<ProgramInputs> for MemAdviceProvider {
     }
 }
 
+impl MemAdviceProvider {
+    // CONSTRUCTORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Creates a new memory advice provider with empty values and sets.
+    pub fn empty() -> Self {
```

Comment:
Question: Any reason not to implement default here and `ProgramInputs` instead of empty?

Reply:
It could be, but I often find `Default` to be a bit ambiguous. The main reason to use `empty` here is to make it clear that it is an empty set.

The `Default` is meant to provide a "minimum usable instance", and that is not necessarily an empty instance. However, in this case, the minimum usable does equal an empty instance, so it is possible.

Its just a matter of preference. I used `empty` here because we had `none` for `ProgramInputs` before (which I renamed to `empty` in this PR to keep it consistent with the other structs), but a `Default` would work just fine too. I usually lean to minimize as much as possible any potential ambiguity, but here is a super weak preference. idk; wdyt?

Reply:
Yeah, makes sense. Thanks.